src/main/java/org/testng/TestRunner.java

```diff
@@ -1557,8 +1577,8 @@ public void onConfigurationSuccess(ITestResult itr) {
     }
   }
 
-  public void setMethodInterceptor(IMethodInterceptor methodInterceptor) {
```

Comment:
Maybe removing the method will break something somewhere. 

The method should be deprecated first and call `addMethodInterceptor` instead, what do you think @cbeust?